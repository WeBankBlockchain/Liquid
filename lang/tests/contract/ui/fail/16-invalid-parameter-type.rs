use liquid_lang as liquid;

#[liquid::contract]
mod noop {
    #[liquid(storage)]
    struct Noop {}

    #[liquid(methods)]
    impl Noop {
        pub fn new(&mut self) {}

        pub fn noop(&self, value: f32) {}
    }
}

fn main() {}
