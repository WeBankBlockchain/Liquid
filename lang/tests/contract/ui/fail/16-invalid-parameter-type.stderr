error[E0277]: the trait bound `f32: You_Should_Use_An_Valid_Input_Type` is not satisfied
  --> $DIR/16-invalid-parameter-type.rs:12:35
   |
12 |         pub fn noop(&self, value: f32) {}
   |                                   ^^^ the trait `You_Should_Use_An_Valid_Input_Type` is not implemented for `f32`
